aws/service.go

```diff
@@ -1767,6 +1780,61 @@ func getSessionWithMaxRetries(ctx context.Context, d *plugin.QueryData, region s
 	return sess, nil
 }
 
+// checkAWSCallerIdent returns boolean if currently logged in to AWS CLI
+func checkAWSCallerIdent(ctx context.Context, profile string) bool {
+	plugin.Logger(ctx).Trace("getSessionWithMaxRetries", "checkAWSCallerIdent", "Starting for "+profile)
+	commandInput := strings.Fields("aws sts get-caller-identity --profile " + profile)
```

Comment:
If the user doesn't have the proper STS permissions to run this particular command, do you know if `aws sso login` would still work as expected? I'm concerned that if a user has set themselves up to use `aws sso login` but not any STS commands, this command would fail.

Reply:
If you don't have St's permission the command will false  return false and the code would assume you logger out as it can't verify it.  Given STS is the AWS best practice way to check if your logged in. I think you trying to code for a bad practice that is going against aws standards when you ask if STS permissions are missing.